src/engine/Server.cpp

```diff
@@ -67,7 +69,56 @@ Awaitable<void> Server::process(
   ad_utility::Timer requestTimer;
   requestTimer.start();
 
-  auto filenameAndParams = ad_utility::UrlParser::parseTarget(request.target());
+  // Parse the payload from a GET or POST request.
+  LOG(DEBUG) << "Request method: \"" << request.method() << "\""
+             << ", target: \"" << request.target() << "\""
+             << ", body: \"" << request.body() << "\"" << std::endl;
+  auto filenameAndParams = [&request]() {
```

Comment:
This lambda (logic + error handling + Comments) has become so long that it can become a separate (static) function of the `Server` class,
As it also takes only one parameter (the `request`).